### DISCUSSION TITLE
Chinese algorithms

### DISCUSSION THREAD

**Discussion by Thermi** [strongswan/strongswan#249] [Ideas]
There's a repo with code for chinese algos: https://github.com/xiaosaa/strongswan/commit/763400d5ffa2279213258fbf32a4d2a3c39b6ed1

Nicely some files already have copyright assigned to Tobias. :laughing:

**Comment by strongX509** [comment]
Hi Thermi,
the code looks quite complete, but there are the following issues to consider:
- There are currently no official IANA IKEv2 parameters defined for the Chinese algorithms, so codepoints from the private use space would have to be selected which in turn would require the sending a Vendor ID (which one?).
- The origin of the actual implementation of the crypto algorithms is unknown. Are they open source and if yes could they be included in strongSwan under an MIT license?
- The code implements the use of the SM1..SM4 crypto algorithms with IKEv1 only, so IKEv2 support would have to be added. This would not be too much of an effort, though.

Did you have any direct contact with the code authors? Before we would go into this we'd have to shed some light on the license issue.

**Reply by Thermi** [reply]
Hi, sure, that would all be required. I just found the repo while going through the network. I don't have any contact with the authors.
